### ISSUE TITLE
CoreOS 1153.4.0 unable to join AWS ECS cluster

### ISSUE THREAD

**Issue by jlambert121** [coreos/bugs#1574]
# Issue Report #

## Bug ##

After updating a CoreOS stable (1122.2.0) host to beta (1153.4.0) the ECS agent is unable to retrieve metadata from Amazon's metadata service.

### CoreOS Version ###

CoreOS 1153.4.0

### Environment ###

AWS t2.micro

### Expected Behavior ###

```
[INFO] Registration completed successfully. I am running as 'arn:aws:ecs:us-east-1:<accountid>:container-instance/1bc43962-62b2-458f-9b12-39d801b89def' in cluster 'ECS'
```

### Actual Behavior ###

```
2016-09-16T11:00:00Z [INFO] Starting Agent: Amazon ECS Agent - v1.12.2 (ecda8a6)
2016-09-16T11:00:00Z [INFO] Loading configuration
2016-09-16T11:00:01Z [WARN] Error accessing the EC2 Metadata Service; retrying: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: i/o timeout
2016-09-16T11:00:03Z [WARN] Error accessing the EC2 Metadata Service; retrying: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: i/o timeout
2016-09-16T11:00:03Z [WARN] Error accessing the EC2 Metadata Service; retrying: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: getsockopt: no route to host
2016-09-16T11:00:04Z [WARN] Error accessing the EC2 Metadata Service; retrying: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: i/o timeout
2016-09-16T11:00:05Z [WARN] Error accessing the EC2 Metadata Service; retrying: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: i/o timeout
2016-09-16T11:00:05Z [CRITICAL] Unable to communicate with EC2 Metadata service to infer region: Get http://169.254.169.254/2014-02-25/dynamic/instance-identity/document: dial tcp 169.254.169.254:80: i/o timeout
2016-09-16T11:00:05Z [CRITICAL] Configuration key not set, key: AWSRegion
2016-09-16T11:00:05Z [CRITICAL] Error loading config: <nil>
```

### Reproduction Steps ###

After booting a stable instance and verifying it correctly worked with the ECS cluster (tasks were deployed through the ECS scheduler), I

```
echo "GROUP=beta" > /etc/coreos/update.conf
sudo systemctl restart update-engine
update_engine_client -update
reboot
```

### Other Information ###

I thought this may be related to #1554, but this happens every time while #1554 appears to be sporadic.  I have tried restarting the docker service and have verified the AWS metadata service is functional by `curl`ing the failed path above from the host.

**Comment by DeviaVir** [comment]
We are testing `1153.4.0` on our cluster as well, and are actually noticing that AWS Instance Profile Roles do not seem to propagate correctly with this version anymore. 

i.e. we are seeing the following response on AWS requests:
```
Unable to locate credentials. You can configure credentials by running "aws configure".
```
while using instance profiles. I assume ECS uses the same setup, which is the problem you are facing here.

cc'ing @crawford as this should probably be seen as a major blocker, it can be expected that many coreos consumers use AWS IAM Roles.

**Comment by jlambert121** [comment]
This looks to be fixed in 1180.0.0.
